opal/datatype/opal_datatype_unpack.c

```diff
@@ -120,7 +124,11 @@ int32_t opal_unpack_homogeneous_contig_function(opal_convertor_t *pConv, struct
             user_memory = pConv->pBaseBuf + pData->true_lb + stack[0].disp + stack[1].disp;
             pConv->bConverted += remaining; /* how much will get unpacked this time */
 
+#if OPAL_ENABLE_DEBUG
```

Comment:
I'm all for getting rid of warnings but not in a way that decreases the readability of the code. Someone will one day wonder why we have such an ugly code in an [already complicated] code.

Reply:
Ok, per conversation on the other comment, I'll update this one with the unused attribute.